components/experimental/src/measure/parser.rs

```diff
@@ -217,13 +218,14 @@ impl MeasureUnitParser {
             };
         }
 
+        // TODO: shall we allow units without any single units?
         // There is no unit without any valid single units.
         if single_units.is_empty() {
             return Err(InvalidUnitError);
         }
 
         Ok(MeasureUnit {
-            single_units,
+            single_units: SingleUnitVec::from_iter(single_units),
```

Comment:
the `single_units` variable here is still a `SmallVec`; we should construct a `SingleUnitVec` directly. You'll have to implement `push` instead of `from_iter`

Reply:
done